maven-core/src/main/java/org/apache/maven/project/artifact/MavenMetadataSource.java

```diff
@@ -96,7 +99,11 @@
     private RepositoryMetadataManager repositoryMetadataManager;
 
     @Requirement
-    private ArtifactFactory repositorySystem;
+    private RepositorySystem repositorySystem;
```

Comment:
this is deprecated becouse only impl we have in maven-compat

Reply:
Can you suggest what is the modern alternative for this? I see it still used in maven-3.9x code base (and actually adapted from there the usage here ...)